src/lpython/semantics/python_comptime_eval.h

```diff
@@ -212,7 +212,7 @@ struct PythonIntrinsicProcedures {
             int64_t a = ASR::down_cast<ASR::IntegerConstant_t>(arg1)->m_n;
             int64_t b = ASR::down_cast<ASR::IntegerConstant_t>(arg2)->m_n;
             return ASR::down_cast<ASR::expr_t>(
-                ASR::make_IntegerConstant_t(al, loc, a%b, type));
+                ASR::make_IntegerConstant_t(al, loc, ((a%b)+b)%b, type));
```

Comment:
Hi, I have handled the compile time evaluations in my pr #1566 and got reviews on it. Could you please remove it from here to avoid clashes.
Also `_mod` function in `lpython_builtin.py` works on runtime evaluations, something like this -
```
def f():
    a:i32 = 8
    b:i32 = -3
    print(a%b)

f()
```
I hope you are aware of that, since all the test cases you refer in the OP are not in that format. Checkout if outputs of the `print(a%b)` format are actually getting corrected by the changes you made in `lpython_builtin.py` . On face value I can see that the below case is handled incorrectly - 
```
def f():
    a:i32 = 8
    b:i32 = 0
    print(a%b)

f()
(lp) C:\Users\kunni\lpython>python try.py
Traceback (most recent call last):
  File "C:\Users\kunni\lpython\try.py", line 6, in <module>
    f()
  File "C:\Users\kunni\lpython\try.py", line 4, in f
    print(a%b)
ZeroDivisionError: integer division or modulo by zero

(lp) C:\Users\kunni\lpython>src\bin\lpython try.py
8
```